src/compiler.ts

```diff
@@ -6681,17 +6681,12 @@ export class Compiler extends DiagnosticEmitter {
             if (instanceMembers && instanceMembers.has(instance.declaration.name.text)) {
               continue; // skip those not inheriting
             }
-            if (mostRecentInheritanceMapping.has(extender)) {
-              let currentRecentParent = assert(mostRecentInheritanceMapping.get(extender));
-              if (currentRecentParent.extends(classInstance)) {
-                // already find recent parent, keep old stmt
-              } else {
-                mostRecentInheritanceMapping.set(extender, classInstance);
-                builder.replaceCase(extender.id, stmts);
-              }
-            } else {
+            if (
+              !mostRecentInheritanceMapping.has(extender) ||
+              !assert(mostRecentInheritanceMapping.get(extender)).extends(classInstance)
```

Comment:
`assert` is unnecessary here due to `has` check before access to entry

Reply:
But ts type checker need this to suppress diagnose.

Reply:
Yes, but is it same as assert?

Reply:
No, it's the same as removing `assert`.

Reply:
AS will report this info, I think it does not worth to use ! instead of assert since I cannot find big benefit from it.
```
INFO AS210: Expression is never 'null'.
      :
 6686 │ !mostRecentInheritanceMapping.get(extender)!.extends(classInstance)
      │  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      └─ in src/compiler.ts(6686,16)
```

Reply:
Fair enough.

Reply:
One workaround is using a `changetype`. That's basically a no-op in asc and a type hint for tsc.

Reply:
That does unfortunately seem clunkier than an `assert` though